src/main/kotlin/com/deviceinsight/helm/DeployMojo.kt

```diff
@@ -51,200 +45,219 @@ class DeployMojo : AbstractMojo(), ServerAuthentication {
+	@Parameter(property = "chartRepoName", required = false)
+	private var chartRepoName: String? = null
 
-	/**
-	 * Name of the chart
-	 */
-	@Parameter(property = "chartName", required = false)
-	private var chartName: String? = null
+	@Parameter(property = "chartRegistryUrl", required = false)
+	private var chartRegistryUrl: String? = null
 
-	@Parameter(property = "chartVersion", required = false)
-	private var chartVersion: String? = null
-
-	@Parameter(property = "chartRepoUrl", required = false)
-	private var chartRepoUrl: String? = null
-
-	@Parameter(property = "chartPublishUrl", required = false)
-	private var chartPublishUrl: String? = null
-
-	@Parameter(property = "chartPublishMethod", required = false, defaultValue = "POST")
-	private var chartPublishMethod: String = "POST"
-
-	@Parameter(property = "chartDeleteUrl", required = false)
-	private var chartDeleteUrl: String? = null
-
-	@Parameter(property = "chartRepoUsername", required = false)
-	private var chartRepoUsername: String? = null
-
-	@Parameter(property = "chartRepoPassword", required = false)
-	private var chartRepoPassword: String? = null
-
-	@Parameter(property = "chartRepoServerId", required = false)
-	private var chartRepoServerId: String? = null
+	@Parameter(property = "deployAtEnd", defaultValue = "false")
+	private var deployAtEnd: Boolean = false
 
 	@Parameter(property = "skipSnapshots", required = false, defaultValue = "true")
 	private var skipSnapshots: Boolean = true
 
-	@Parameter(defaultValue = "\${project}", readonly = true, required = true)
-	private lateinit var project: MavenProject
 
 	@Parameter(defaultValue = "\${reactorProjects}", required = true, readonly = true)
 	private lateinit var reactorProjects: List<MavenProject>
 
-	@Parameter(defaultValue = "\${settings}", readonly = true)
-	override lateinit var settings: Settings
-
-	@Parameter(property = "helm.skip", defaultValue = "false")
-	private var skip: Boolean = false
-
-	@Parameter(property = "helm.deployAtEnd", defaultValue = "false")
-	private var deployAtEnd: Boolean = false
-
-	@Throws(MojoExecutionException::class)
-	override fun execute() {
+	override fun runMojo() {
+		check(chartRepoName == null || chartRegistryUrl == null) {
+			"chartRepoName and chartRegistryUrl must not be set at the same time"
+		}
 
-		if (skip) {
-			log.info("helm-deploy has been skipped")
-			return
+		if (chartRepoName == null && chartRegistryUrl == null) {
+			// If there is only one repo or registry specified we autodetect the name/url property
+
+			if (repos.size == 1 && registries.isEmpty()) {
+				chartRepoName = repos.first().name
+			} else if (registries.size == 1 && repos.isEmpty()) {
+				chartRegistryUrl = registries.first().url
+			} else if (registries.isEmpty() && repos.isEmpty()) {
+				error(
+					"It is necessary to specify chartRepoName or chartRegistryUrl explicitly " +
+						"since there is no chart repo / chart registry configured"
```

Comment:
Sorry, I don't understand the error message – what good is specifying a chartRepoName when no chart repo is configured?